lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,11 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp check_lockfile_conflicts(lockfile, info) do
+    if Regex.run(~r/<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|/, info) do
```

Comment:
If we match on the beginning of the line, a regex seems simpler, if we don't, `String.contains?` should indeed be just fine. What do you think?